pkg/resourcemanager/cosmosdb/sqldatabase/cosmosdbsql.go

```diff
@@ -0,0 +1,294 @@
+		ctx,
+		instance.Spec.ResourceGroup,
+		instance.Spec.Account,
+		instance.ObjectMeta.Name)
+	if err != nil {
+		azerr := errhelp.NewAzureError(err)
+		if azerr.Type != errhelp.ResourceNotFound && azerr.Type != errhelp.NotFoundErrorCode {
+			instance.Status.Message = fmt.Sprintf("failed GET-ing CosmosDBSQLDatabase: %s", err.Error())
+			return errhelp.IsErrorFatal(err, notFoundErrorCodes, nil)
+		}
+	}
+
+	// TODO: There's a bug in Cosmos DB SQL Database where the throughput settings configured on
+	// TODO: the DB don't actually get shown on the DB, instead they are shown on the throughput child resource.
+	// TODO: We monitor that resource too so we can diff against it
+	sqlDatabaseThroughputSettings, err := cosmosDBSQLDatabaseClient.GetSQLDatabaseThroughput(
+		ctx,
+		instance.Spec.ResourceGroup,
+		instance.Spec.Account,
+		instance.ObjectMeta.Name)
+	if err != nil {
+		azerr := errhelp.NewAzureError(err)
+		if azerr.Type != errhelp.ResourceNotFound && azerr.Type != errhelp.NotFoundErrorCode {
+			instance.Status.Message = fmt.Sprintf("failed GET-ing CosmosDBSQLDatabaseThroughput: %s", err.Error())
+			return errhelp.IsErrorFatal(err, notFoundErrorCodes, nil)
+		}
+	}
+
+	// Transform our spec to Azure shape
+	transformed := m.transformToCosmosDBSQLDatabase(instance)
+
+	// Check if we're in the goal state already
+	resourceMatchesAzure := DoesResourceMatchAzure(transformed, sqlDatabase, sqlDatabaseThroughputSettings)
+	if resourceMatchesAzure {
+		instance.Status.SetProvisioned(resourcemanager.SuccessMsg)
+		instance.Status.ResourceId = *sqlDatabase.ID
+		return true, nil
+	}
+
+	// Drive to goal state
+	instance.Status.SetProvisioning("")
+	future, err := cosmosDBSQLDatabaseClient.CreateUpdateSQLDatabase(
+		ctx,
+		instance.Spec.ResourceGroup,
+		instance.Spec.Account,
+		instance.ObjectMeta.Name,
+		transformed)
+	if err != nil {
+		instance.Status.Message = err.Error()
+		allowedErrors := []string{
+			errhelp.AsyncOpIncompleteError,
+			errhelp.AlreadyExists,
+			errhelp.FailoverGroupBusy,
+		}
+		unrecoverableErrors := []string{
+			errhelp.InvalidFailoverGroupRegion,
+		}
+		return errhelp.IsErrorFatal(err, append(allowedErrors, notFoundErrorCodes...), unrecoverableErrors)
+	}
+	instance.Status.SetProvisioning("Resource request successfully submitted to Azure")
+	instance.Status.SetPollingURL(future.PollingURL(), api.PollingURLKindCreateOrUpdate)
+
+	// Need to poll the polling URL, so not done yet!
+	return false, nil
+}
+
+func (m *AzureCosmosDBSQLDatabaseManager) Delete(ctx context.Context, obj runtime.Object, opts ...resourcemanager.ConfigOption) (bool, error) {
+	instance, err := m.convert(obj)
+	if err != nil {
+		return false, err
+	}
+
+	cosmosDBSQLDatabaseClient, err := cosmosdb.GetCosmosDBSQLDatabaseClient(m.Creds)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to create cosmos DB SQL database client")
+	}
+
+	// TODO: What if we have a pollURL already but it's for a create
```

Comment:
I think you've handled this now, right? I guess you were typing this and then thought "darn it, now I have to go fix that!".

Reply:
Oh yea - will fix in a follow-on PR though just to not invalidate my CI pass